build/__tools__/run-tests-with-database.sh

```diff
@@ -0,0 +1,189 @@
+
+
+try pdo-test-request --no-ledger   \
+    --eservice-name e1 --logfile $PDO_HOME/logs/client.log --loglevel info --eservice-db ${ESERVICE_DB}
+
+try pdo-test-contract --no-ledger --contract integer-key \
+    --eservice-name e2 --logfile $PDO_HOME/logs/client.log --loglevel info --eservice-db ${ESERVICE_DB}
+
+# make sure we have the necessary files in place
+CONFIG_FILE=${PDO_HOME}/etc/pcontract.toml
+if [ ! -f ${CONFIG_FILE} ]; then
+    die missing client configuration file, ${CONFIG_FILE}
+fi
+
+CONTRACT_FILE=${PDO_HOME}/contracts/_mock-contract.scm
+if [ ! -f ${CONTRACT_FILE} ]; then
+    die missing contract source file, ${CONTRACT_FILE}
+fi
+
+try pdo-create --config ${CONFIG_FILE} --ledger ${PDO_LEDGER_URL} \
+     --logfile $PDO_HOME/logs/client.log --loglevel info \
+    --identity user1 --save-file ${SAVE_FILE} \
+    --contract mock-contract --source _mock-contract.scm --eservice-name e1 e2 e3 --eservice-db ${ESERVICE_DB}
+
+# this will invoke the increment operation 5 times on each enclave round robin
+# fashion; the objective of this test is to ensure that the client touches
+# multiple, independent enclave services and pushes missing state correctly
+declare -i pcontract_es=3 #.see ../opt/pdo/etc/template/pcontract.toml
+declare -i n=$((NUM_SERVICES*pcontract_es)) e v value
+for v in $(seq 1 ${n}) ; do
+    e=$((v % pcontract_es + 1))
+    value=$(pdo-update --config ${CONFIG_FILE} --ledger ${PDO_LEDGER_URL} \
+                       --eservice-name e${e} \
+                       --logfile $PDO_HOME/logs/client.log --loglevel info \
+                       --identity user1 --save-file ${SAVE_FILE} --eservice-db ${ESERVICE_DB} \
+                       "'(inc-value)")
+    if [ $value != $v ]; then
+        die "contract has the wrong value ($value instead of $v) for enclave $e"
+    fi
+done
+
+KEYGEN=${SRCDIR}/build/__tools__/make-keys
+if [ ! -f ${PDO_HOME}/keys/red_type_private.pem ]; then
+    for color in red green blue ; do
+        ${KEYGEN} --keyfile ${PDO_HOME}/keys/${color}_type --format pem
+        ${KEYGEN} --keyfile ${PDO_HOME}/keys/${color}_vetting --format pem
+        ${KEYGEN} --keyfile ${PDO_HOME}/keys/${color}_issuer --format pem
+    done
+fi
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+try pdo-shell --logfile $PDO_HOME/logs/client.log --loglevel info --ledger $PDO_LEDGER_URL \
+    --eservice-name e1 e2 e3 e4 e5 --eservice-db ${ESERVICE_DB} -s ${SRCDIR}/contracts/exchange/scripts/create.psh -m color red 
+
+for p in $(seq 1 5); do
+    pdo-shell --logfile $PDO_HOME/logs/client.log --loglevel info --ledger $PDO_LEDGER_URL \
+    --eservice-name e${p} --eservice-db ${ESERVICE_DB} -s ${SRCDIR}/contracts/exchange/scripts/issue.psh -m color red -m issuee user$p -m count $(($p * 10))
+done
+
+yell completed all tests
+exit 0
```

Comment:
My big question is why is this a separate file? Just incorporate your tests into the main run-tests.

Reply:
sure will do, i was only afraid of making it too big